### PR TITLE
Scrape store products

### PR DESCRIPTION
**Pull request by gwsii**
Provides some basic automation to push product/create events onto our RetailStream.

Includes serverless.yml which import roles from the retail/stream part of the project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,51 +1,6 @@
...
-.grunt
-
-# node-waf configuration
-.lock-wscript
-
-# Compiled binary addons (http://nodejs.org/api/addons.html)
-build/Release
-
-# Dependency directories
+# package directories
```
Comment (erikerikson) [inline .gitignore]: why is so much deleted?

File: products/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ 
+```JSON
+{
+  "productFrequency": 1000,
+  "lambdaTimeout": 30000
+}
+```
+
+Where `productFrequency` is the interval between writes of products to the Kinesis stream and `lambdaTimeout`
+is the duration for which the process will run.
```
Comment (erikerikson) [inline products/README.md]: units?

File: products/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+This is a Serverless project, so the [`serverless`](http://serverless.com) command must be installed:
+
+`npm install `
```
Comment (erikerikson) [inline products/README.md]: `npm install serverless`?

File: products/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+To deploy the service, use:
+
+`serverless deploy`
```
Comment (erikerikson) [inline products/README.md]: `sls`?

File: products/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+The `event.json` file contains the JSON to control product production frequency `productFrequency` and 
+run duration `lambdaTimeout` for running the Lambda locally with this command:
+  
+`node produce-products-exec.js`
```
Comment (erikerikson) [inline products/README.md]: did you look into `sls invoke local`?

File: products/event.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "productFrequency": 1000,
+  "lambdaTimeout": 3600000
```
Comment (erikerikson) [inline products/event.json]: wouldn't this value timeout if running in the context of lambda?

File: retail/stream/serverless.yml [hunk 7]
```diff
@@ -0,0 +1,99 @@
+frameworkVersion: '>=1.0.0 <2.0.0'
```
Comment (erikerikson) [inline retail/stream/serverless.yml]: this file should not be here

File: products/event.json (hunk 6 above)
Comment (gwsii) [inline products/event.json]: Yes. The purpose of this event.json is to allow for the loacal execution and is meant to be long-running.

File: products/README.md (hunk 2 above)
Comment (gwsii) [inline products/README.md]: Fixed.

**Comment by erikerikson** [comment]
Please re-target the initial-formation branch since the code this relates to is being worked there.  If we'd like to change our workflow, lets talk about that.

**Comment by gwsii** [comment]
Going to merge into WIP branch: initial-formation.

**Pull request closed without merging**